src/omv/sensor.c

```diff
@@ -524,15 +527,32 @@ int sensor_write_reg(uint16_t reg_addr, uint16_t reg_data)
 
 int sensor_set_pixformat(pixformat_t pixformat)
 {
-    uint32_t jpeg_mode = DCMI_JPEG_DISABLE;
+    if ((pixformat == PIXFORMAT_JPEG)
+    && (MAIN_FB()->x // non-zero
+    ||  MAIN_FB()->y // non-zero
+    || (MAIN_FB()->u != resolution[sensor.framesize][0]) // it was changed
+    || (MAIN_FB()->v != resolution[sensor.framesize][1]) // it was changed
+    || sensor.transpose
+    || sensor.auto_rotation)) {
+        return -1;
+    }
 
-    if (sensor.pixformat == pixformat) {
+    // sensor_check_buffsize() will switch from PIXFORMAT_BAYER to PIXFORMAT_RGB565 to try to fit
+    // the MAIN_FB() in RAM as a first step optimization. If the user tries to switch back to RGB565
+    // and that would be bigger than the RAM buffer we would just switch back.
+    //
+    // So, just short-circuit doing any work.
+    if ((sensor.pixformat == PIXFORMAT_BAYER)
+    &&  (pixformat == PIXFORMAT_RGB565)
+    &&  (MAIN_FB()->u * MAIN_FB()->v * 2 > OMV_RAW_BUF_SIZE)
+    &&  (MAIN_FB()->u * MAIN_FB()->v * 1 <= OMV_RAW_BUF_SIZE)) {
```

Comment:
No, this has to be put here. I added this code for the explicit purpose so that if you proceed to call:

`sensor.set_pixformat(sensor.RGB565)
sensor.set_framesize(sensor.VGA)
sensor.snapshot()`

In a loop you don't re-trigger changing the frame each loop. I ran into this issue when developing the RPC library. The reason for me even for starting this PR was to deal with this problem.

Reply:
Added a comment.